src/resources/Cookie/CookieMethods.ts

```diff
@@ -13,3 +15,22 @@ export function getCookie(cname: string) {
   }
   return '';
 }
+
+export const useCookieListener = (cookieName: string, interval = 2000) => {
+  const [cookie, setCookie] = useState(getCookie(cookieName));
+  const [displayAlert, setDisplayAlert] = useState(false);
+
+  useEffect(() => {
+    const checkCookie = setInterval(() => {
+      const currentCookie = getCookie(cookieName);
+      if (currentCookie !== cookie) {
+        setCookie(currentCookie);
+        setDisplayAlert(true);
+      }
+    }, interval); // Check every interval
+
+    return () => clearInterval(checkCookie);
+  }, [cookie, cookieName]);
```

Comment:
I suggest not having the cookie in the dependency array. We use setCookie inside the interval, and whenever the cookie state changes, the effect will re-run, potentially causing issues. 

```suggestion
  }, [cookie, interval]);
```

Reply:
Good point - I'll fix! 👩‍🔧